test/galaxy_selenium/navigates_galaxy.py

```diff
@@ -148,14 +148,19 @@ def history_has_hid(driver):
         contents = self.api_get("histories/%s/contents" % current_history_id)
         history_item = [d for d in contents if d["hid"] == hid][0]
         history_item_selector = "#%s-%s" % (history_item["history_content_type"], history_item["id"])
-        try:
-            self.wait_for_selector_visible(history_item_selector)
-        except self.TimeoutException as e:
-            dataset_elements = self.driver.find_elements_by_css_selector("#current-history-panel .list-items div")
-            div_ids = [d.get_attribute('id') for d in dataset_elements]
-            template = "Failed waiting on history item %d to become visible, visible datasets include [%s]."
-            message = template % (hid, ",".join(div_ids))
-            raise self.prepend_timeout_message(e, message)
+        for i in range(3):
```

Comment:
Should this be configurable with the allow_refresh param?

Reply:
Yeah... my logic is a little off here isn't it?

Reply:
Or I guess you are saying this should be parameterized instead of 3 - yeah I'll make a couple quick changes to this.